oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/DefaultOAuth2User.java

```diff
@@ -57,12 +57,15 @@ public class DefaultOAuth2User implements OAuth2User, Serializable {
 
 	private final String nameAttributeKey;
 
+	private final String name;
+
 	/**
 	 * Constructs a {@code DefaultOAuth2User} using the provided parameters.
 	 * @param authorities the authorities granted to the user
 	 * @param attributes the attributes about the user
 	 * @param nameAttributeKey the key used to access the user's &quot;name&quot; from
 	 * {@link #getAttributes()}
+	 * @deprecated
 	 */
 	public DefaultOAuth2User(Collection<? extends GrantedAuthority> authorities, Map<String, Object> attributes,
```

Comment:
Can this constructor be changed to call the new constructor? This may simplify future maintenance.

Reply:
@jzheaux, doing so caused `NullPointerException`s, because I am getting the `name` out of `attributes` map before checking for nulls in the new constructor.

Reply:
another solution would be by introducing an `init` method to encapsulate common logic, but will require all `authorities`, `attributes`, and `name` to be **non final**.

init method would look something like
```
private void init(Map<String, Object> attributes, Collection<? extends GrantedAuthority> authorities,
					  String name) {
        this.authorities = (authorities != null)
		        ? Collections.unmodifiableSet(new LinkedHashSet<>(this.sortAuthorities(authorities)))
		        : Collections.unmodifiableSet(new LinkedHashSet<>(AuthorityUtils.NO_AUTHORITIES));
        this.attributes = Collections.unmodifiableMap(new LinkedHashMap<>(attributes));
        this.name = name;
}
```
old constructor
```
public DefaultOAuth2User(Collection<? extends GrantedAuthority> authorities, Map<String, Object> attributes,
			String nameAttributeKey) {
        Assert.notEmpty(attributes, "attributes cannot be empty");
        Assert.hasText(nameAttributeKey, "nameAttributeKey cannot be empty");
        if (!attributes.containsKey(nameAttributeKey)) {
	        throw new IllegalArgumentException("Missing attribute '" + nameAttributeKey + "' in attributes");
        }
        init(attributes, authorities,(String) attributes.get(nameAttributeKey));
}
```
and new constructor
```
public DefaultOAuth2User(Map<String, Object> attributes, Collection<? extends GrantedAuthority> authorities,
			String name) {
        Assert.notEmpty(attributes, "attributes cannot be empty");
        Assert.hasText(name, "name cannot be empty");
        init(attributes, authorities, name);
}
```